xbmc/filesystem/File.cpp

```diff
@@ -254,6 +254,19 @@ bool CFile::Open(const std::string& strFileName, const unsigned int flags)
 
 bool CFile::Open(const CURL& file, const unsigned int flags)
 {
+  if (m_pFile)
+  {
+    if ((flags & READ_REOPEN) == 0)
+    {
+      CLog::Log(LOGERROR, "File::Open - already open: %s", file.GetRedacted().c_str());
+      throw;
```

Comment:
@FernetMenta I suggested it for that reason, as this code path is a developer error and not runtime error it helps catch mistakes early. I prefer an uncaught throw over terminate as throw will break into debugger or trigger a crash dump.